fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/impl/transfer/item/SidedInventorySlotWrapper.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.transfer.item;
+
+import net.minecraft.inventory.SidedInventory;
+import net.minecraft.util.math.Direction;
+
+import net.fabricmc.fabric.api.transfer.v1.item.ItemVariant;
+import net.fabricmc.fabric.api.transfer.v1.storage.base.SingleSlotStorage;
+import net.fabricmc.fabric.api.transfer.v1.transaction.TransactionContext;
+
+/**
+ * Wrapper around an {@link InventorySlotWrapper}, with additional canInsert and canExtract checks.
+ */
+class SidedInventorySlotWrapper implements SingleSlotStorage<ItemVariant> {
+	private final InventorySlotWrapper slotWrapper;
+	private final SidedInventory sidedInventory;
+	private final Direction direction;
+
+	SidedInventorySlotWrapper(InventorySlotWrapper slotWrapper, SidedInventory sidedInventory, Direction direction) {
+		this.slotWrapper = slotWrapper;
+		this.sidedInventory = sidedInventory;
+		this.direction = direction;
+	}
+
+	@Override
+	public long insert(ItemVariant resource, long maxAmount, TransactionContext transaction) {
+		if (!sidedInventory.canInsert(slotWrapper.slot, resource.toStack(), direction)) {
+			return 0;
+		} else {
+			return slotWrapper.insert(resource, maxAmount, transaction);
+		}
+	}
+
+	@Override
+	public long extract(ItemVariant resource, long maxAmount, TransactionContext transaction) {
+		if (!sidedInventory.canExtract(slotWrapper.slot, resource.toStack(), direction)) {
+			return 0;
+		} else {
+			return slotWrapper.insert(resource, maxAmount, transaction);
```

Comment:
This is suspicious, shouldn't it be `extract`?

Reply:
Copy pasta...

Reply:
Fixed, thanks :smile: